server/worker/docker/docker.go

```diff
@@ -62,6 +62,14 @@ func (d *Docker) Do(c context.Context, r *worker.Work) {
 	// mark the build as Started and update the database
 	r.Commit.Status = model.StatusStarted
 	r.Commit.Started = time.Now().UTC().Unix()
+
+	// Update the sequential build number if it hasn't been set yet
+	if r.Commit.BuildNumber == 0 {
```

Comment:
I think the ability to increment should live in the `database.PostCommit` function instead.